### PR TITLE
Improve the Docker Image Helper Script

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This PR updates the ` generate_cyhy_docker_image.sh` helper script to:
1. Pull the MaxMind GeoIP2 license key from AWS SSM's Parameter Store.
1. Allow you to specify a custom image name without having to specify a tag.
1. Update the output file name to reflect the image name and tag being used.
1. Improve the way the script handles command line arguments, making it easier to add additional functionality in the future.

## 💭 Motivation and Context ##

After a team discussion I needed to change how the MaxMind license key was retrieved to reflect the way I was doing it in my local repository. Pulling directly from the Parameter Store using the AWS CLI is a better way to populate that value than how it was done previously. While I was updating the script I wanted to add the ability to specify a custom image name without needed to specify a custom tag. This necessitated rewriting the command line argument logic.

## 🧪 Testing ##

I verified that the command line arguments work as expected.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] All relevant type-of-change labels have been added.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
* [x] All new and existing tests pass.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,25 +90,39 @@ public Docker Hub if it was built with your MaxMind license key!**
...
+```
 
 #### Default name and tag
 ```console
 ./generate_cyhy_docker_image.sh
 ```
 
+#### Specified name and default tag
+```console
+./generate_cyhy_docker_image.sh --image-name cisagove/cyhy-env
```
Comment (jsf9k) [inline README.md]: Is "cisagove" a typo?

**Review by jsf9k — APPROVED** [review]
Looks good to me, except for one tiny typo.

File: README.md (hunk 1 above)
Comment (mcdonnnj) [reply README.md]: Yes, yes it is. Glad you noticed!

Comment (mcdonnnj) [reply README.md]: Fixed in 572bdd3.

**Review by dav3r — APPROVED** [review]
Great improvements! 🔩

**Pull request merged**